tests-e2e/fixture/application/fixture.go

```diff
@@ -124,3 +125,36 @@ func HaveSyncStatusCode(status appv1alpha1.ApplicationStatus) matcher.GomegaMatc
 		return res
 	}, BeTrue())
 }
+
+//  HaveApplicationSyncError checks the Application .status.conditions.Message fiels is set with syncError.
+func HaveApplicationSyncError(syncError appv1alpha1.ApplicationStatus) matcher.GomegaMatcher {
+
+	return WithTransform(func(app appv1alpha1.Application) bool {
+
+		k8sClient, err := fixture.GetKubeClient()
+		if err != nil {
+			fmt.Println(k8sFixture.K8sClientError, err)
+			return false
+		}
+
+		err = k8sClient.Get(context.Background(), client.ObjectKeyFromObject(&app), &app)
+		if err != nil {
+			fmt.Println(k8sFixture.K8sClientError, err)
+			return false
+		}
+
+		var existingConditionMessage string
+		var res bool
+
+		for _, msg := range app.Status.Conditions {
+			existingConditionMessage = msg.Message
```

Comment:
Should we check if the condition is of type "SyncError" before copying the message? There could be other conditions as well.